### DISCUSSION TITLE
Integration script on every page Goodays

### DISCUSSION THREAD

**Discussion by leopalazzi** [vercel/next.js#65611] [Help]
### Summary

Hello,

I need to integrate the script of goodays but it needs to be loaded on every page where I use it.

I try multiple method (passing a key to my component, using useEffect) but nothing is working.

Can someone help ?

Here is what it looks like on their documentation :
```
<script type="text/javascript">
    (function (i, s, o, g, r, a, m) {
        i[r] = i[r], a = s.createElement(o), m = s.getElementsByTagName(o)[0];
        a.async = 1;
        a.src = g;
        m.parentNode.insertBefore(a, m)
    })(window, document, 'script', 'https://cdn.goodays.co/sdk/<API_KEY>.js', 'cz');
</script>
```

And here is the documentation : [Goodays SDK](https://developers.goodays.co/docs/web-sdk)

**Comment by SouthLink** [comment]
You can add this logic in the layout.tsx component in the app directory or in the _app.tsx component in the pages directory and fill in your <API_KEY>.

**Reply by leopalazzi** [reply]
I tried but this is not loading at every page

**Reply by SouthLink** [reply]
Following this logic should insert the script in all pages, except through the next/router jump, so that it does not reload, but loads when the page is refreshed, or you can try using the [inline script insert method](https://nextjs.org/docs/pages/building-your-application/optimizing/scripts#inline-scripts)

**Reply by leopalazzi** [reply]
Yes I try but the issue is that I need the script to be loaded after the page is rendered.

**Reply by icyJoseph** [reply]
Show us how you are adding it, and is it App Router, Pages Router?

**Reply by leopalazzi** [reply]
@icyJoseph 

I made a component with the script and I am adding it into the pages I need it. I tried also in _app.js and _document.js but nothing is working properly.

```
            <script
                type="text/javascript"
                dangerouslySetInnerHTML={{
                    __html: `(function(i,s,o,g,r,a,m) {i[r]=i[r],a=s.createElement(o),m=s.getElementsByTagName(o)[0];console.log("test");a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,'script', 'https://cdn.goodays.co/sdk/API_KEY.js','cz');`,
                }}
            ></script>
```

**Reply by leopalazzi** [reply]
I tried with that but it's not loading after the render of every page. I am trying to call the function in a `useEffect `
